pkg/cloud/packet/client.go

```diff
@@ -150,27 +153,41 @@ func (p *Client) NewDevice(ctx context.Context, req CreateDeviceRequest) (*packn
 		facility = packetMachineSpec.Facility
 	}
 
-	serverCreateOpts := &packngo.DeviceCreateRequest{
-		Hostname:      req.MachineScope.Name(),
-		ProjectID:     packetClusterSpec.ProjectID,
-		Metro:         metro,
-		BillingCycle:  packetMachineSpec.BillingCycle,
-		Plan:          packetMachineSpec.MachineType,
-		OS:            packetMachineSpec.OS,
-		IPXEScriptURL: packetMachineSpec.IPXEUrl,
-		Tags:          tags,
-		UserData:      userData,
-	}
+	hostname := req.MachineScope.Name()
+
+	serverCreateOpts := metal.CreateDeviceRequest{}
 
 	if facility != "" {
-		serverCreateOpts.Facility = []string{facility}
+		serverCreateOpts.DeviceCreateInFacilityInput = &metal.DeviceCreateInFacilityInput{
+			Hostname:        &hostname,
+			Facility:        []string{facility},
+			BillingCycle:    &req.MachineScope.PacketMachine.Spec.BillingCycle,
+			Plan:            req.MachineScope.PacketMachine.Spec.MachineType,
+			OperatingSystem: req.MachineScope.PacketMachine.Spec.OS,
+			IpxeScriptUrl:   &req.MachineScope.PacketMachine.Spec.IPXEUrl,
+			Tags:            tags,
+			Userdata:        &userData,
+		}
+	} else {
+		serverCreateOpts.DeviceCreateInMetroInput = &metal.DeviceCreateInMetroInput{
+			Hostname:        &hostname,
+			Metro:           metro,
+			BillingCycle:    &req.MachineScope.PacketMachine.Spec.BillingCycle,
+			Plan:            req.MachineScope.PacketMachine.Spec.MachineType,
+			OperatingSystem: req.MachineScope.PacketMachine.Spec.OS,
+			IpxeScriptUrl:   &req.MachineScope.PacketMachine.Spec.IPXEUrl,
+			Tags:            tags,
+			Userdata:        &userData,
+		}
 	}
 
 	reservationIDs := strings.Split(packetMachineSpec.HardwareReservationID, ",")
 
 	// If there are no reservationIDs to process, go ahead and return early
-	if len(reservationIDs) == 0 {
-		dev, _, err := p.Devices.Create(serverCreateOpts)
+	if len(reservationIDs) <= 1 {
+		apiRequest := p.DevicesApi.CreateDevice(ctx, req.MachineScope.PacketCluster.Spec.ProjectID)
+		// nolint:bodyclose // Body is closed inside Execute()
```

Comment:
I'm not sure why the 3 `nolint:bodyclose` comments are needed; there are other `Execute` calls in this file that don't trigger a lint error, and all of the `Execute` methods are generated by the same generator, and include a call to `Body.Close()`.  In this case, the relevant `metal-go` code is: https://github.com/equinix-labs/metal-go/blob/main/metal/v1/api_devices.go#L293-L295

Reply:
Turns out I needed to add `nolint:bodyclose` directives in a few more places, and the linter was only showing me the first 3 violations.  This PR passes lint, and I think this directive is acceptable since the response body is closed inside the called function, but it'd be nice to get rid of these false positives.

Reply:
Relevant issue in the bodyclose linter: https://github.com/timakin/bodyclose/issues/42